### PR TITLE
Fix 2D and 3D memset

### PR DESCRIPTION
**Pull request by gargrahul**

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
From looking to the code it looks good. I will also perform some test within the next days.

From the performance perspective it would make more sende to use a kernel to perform the ND memset instead of having a loop and spawn multiple 1D memsets. IMO such an optimization should be shipped later as separate pull request.

**Comment by psychocoderHPC** [comment]
Fix is working. 
@gargrahul I suggest to add CI test similar you added in #1798

File: src/hip_memory.cpp [hunk 1]
```diff
@@ -2219,16 +2219,40 @@ hipError_t hipMemset(void* dst, int value, size_t sizeBytes) {
...
+    if ((pitch == width) && (height == setHeight)) {
+        sizeBytes = pitch * setHeight * depth;
+        if(async)
+            return ihipMemsetAsync(dst, value, sizeBytes, stream, copyDataType);
+        else
+            return ihipMemsetSync(dst, value, sizeBytes, nullptr, copyDataType);
+    } else {
+        for(size_t i = 0; i < depth; ++i) {
+            for(size_t j = 0; j < setHeight; ++j) {
+                int* dstPtr = (int*) ((unsigned char*) dst + i * height * pitch + j * pitch);
```
Comment (arghdos) [inline src/hip_memory.cpp:2235]: Wont this only work if dst is a 4-byte type?

Comment (arghdos) [inline src/hip_memory.cpp:2235]: Is there any reason to cast this to an int*? Shouldn't it be void*?

Comment (gargrahul) [reply src/hip_memory.cpp]: I have corrected this.

**Pull request merged**